pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -196,8 +198,44 @@ type CanaryStep struct {
 	Pause *RolloutPause `json:"pause,omitempty"`
 	// Experiment defines the experiment object that should be created
 	Experiment *RolloutExperimentStep `json:"experiment,omitempty"`
+	// AnalysisRun defines the AnalysisRun that will run for a step
+	AnalysisRun *RolloutAnalysisRun `json:"analysisRun,omitempty"`
 }
 
+// RolloutAnalysisRun defines a template that is used to create a analysisRun
+type RolloutAnalysisRun struct {
+	// TemplateName reference of the AnalysisTemplate name used by the Rollout to create the run
+	TemplateName string `json:"templateName,omitempty"`
+	// Arguments the arguments that will be added to the AnalysisRuns
+	Arguments []AnalysisRunArgument `json:"arguments,omitempty"`
+}
+
+// AnalysisRunArgument argument to add to analysisRun
+type AnalysisRunArgument struct {
+	// Name argument name
+	Name string `json:"name"`
+	// Value a hardcoded value for the argument. This field is a one of field with valueFrom
+	Value string `json:"value,omitempty"`
+	// ValueFrom A reference to where the value is stored. This field is a one of field with valueFrom
+	ValueFrom *ArgumentValueFrom `json:"valueFrom,omitempty"`
+}
+
+// ArgumentValueFrom defines references to fields within resources to grab for the value (i.e. Pod Template Hash)
+type ArgumentValueFrom struct {
+	// PodTemplateHashValue gets the value from one of the children ReplicaSet's Pod Template Hash
+	PodTemplateHashValue *ValueFromPodTemplateHash `json:"podTemplateHashValue,omitempty"`
+}
+
+// ValueFromPodTemplateHash indicates which ReplicaSet pod template pod hash to use
+type ValueFromPodTemplateHash string
+
+const (
```

Comment:
For consistency, our code words have had the first letter capitalized, so I think this should be `Stable` and `New`. 

Also, debating if the term should be `New`, `Canary`, `Current`?

Reply:
I like `Current` or `New` because of the BlueGreen case and we can reuse this code works.

Reply:
After discussing with @jessesuen, we decided to go with `Latest`